app/src/main/java/ch/epfl/cs311/wanderwave/model/data/Beacon.kt

```diff
@@ -0,0 +1,38 @@
+package ch.epfl.cs311.wanderwave.model.data
+
+import com.google.firebase.firestore.DocumentSnapshot
+
+data class Beacon(
+
+    /** GUID of the beacon */
+    val id: String,
+
+    /** Location of the beacon */
+    val location: Location,
+
+    /** List of tracks that are broadcast from the beacon */
+    val tracks: List<Track> = listOf<Track>(),
+) {
+  fun toMap(): Map<String, Any> {
+    return hashMapOf("id" to id, "location" to location.toMap(), "tracks" to listOf<Track>())
+  }
```

Comment:
Why toes toMap just put an empty track list for "tracks"?

Reply:
@KlaKalma

Reply:
Because it cannot retrieve from the beacon document the track, there are references but not the tracks themself, that need to be fetched, we could add the list of tracks inside the beacon